rust/src/dataset/scanner.rs

```diff
@@ -127,17 +139,24 @@ impl ScannerStream {
             for frag in &fragments {
                 let data_file = &frag.files[0];
                 let path = data_dir.child(data_file.path.clone());
-                let reader =
-                    match FileReader::new(&object_store, &path, Some(manifest.as_ref())).await {
-                        Ok(mut r) => {
-                            r.set_projection(schema.clone());
-                            r
-                        }
-                        Err(e) => {
-                            tx.send(Err(e)).await.unwrap();
-                            continue;
-                        }
-                    };
+                let reader = match FileReader::try_new_with_fragment(
+                    &object_store,
+                    &path,
+                    frag.id,
+                    Some(manifest.as_ref()),
+                )
+                .await
+                {
+                    Ok(mut r) => {
+                        r.set_projection(schema.clone());
+                        r.with_row_id(with_row_id);
+                        r
+                    }
+                    Err(e) => {
+                        tx.send(Err(e)).await.unwrap();
+                        continue;
```

Comment:
does it make sense to keep scanning when there's an error?

Reply:
Good question. There is an TryStream trait tho, https://docs.rs/futures/latest/futures/stream/trait.TryStream.html, maybe we should implement TryStream instead of stream? and stop-if-fail?